### PR TITLE
Adding citation file

### LINKED ISSUE #6700: How to Cite pydantic
### Initial Checks

- [X] I have searched Google & GitHub for similar requests and couldn't find anything
- [X] I have read and followed [the docs](https://docs.pydantic.dev) and still think this feature is missing

### Description

I make it a habit to cite all libraries that I use heavily in my projects in my presentations, papers, resports.... I think that is the least people should do when they utilize such great libraries. For pydantic, i don't see any official documentation on the way to cite it.

It would be great to have it in the main page of github and documentation. 

### Affected Components

- [ ] [Compatibility between releases](https://docs.pydantic.dev/changelog/)
- [ ] [Data validation/parsing](https://docs.pydantic.dev/usage/models/#basic-model-usage)
- [ ] [Data serialization](https://docs.pydantic.dev/usage/exporting_models/) - `.model_dump()` and `.model_dump_json()`
- [ ] [JSON Schema](https://docs.pydantic.dev/usage/schema/)
- [ ] [Dataclasses](https://docs.pydantic.dev/usage/dataclasses/)
- [ ] [Model Config](https://docs.pydantic.dev/usage/model_config/)
- [ ] [Field Types](https://docs.pydantic.dev/usage/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://docs.pydantic.dev/usage/validation_decorator/)
- [ ] [Generic Models](https://docs.pydantic.dev/usage/models/#generic-models)
- [ ] [Other Model behaviour](https://docs.pydantic.dev/usage/models/) - `model_construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://docs.pydantic.dev/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

Selected Assignee: @Kludex

### PR DESCRIPTION
**Pull request by priya-gitTest**
## Related issue number

Fix #6700

## Checklist

* [ ] The pull request title is a good summary of the changes - it will be used in the changelog
* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI
* [ ] Documentation reflects the changes where applicable
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @hramezani

### CODE REVIEW

**Comment by priya-gitTest** [comment]
please review" to assign reviewers

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        family-names: Badaracco
+    identifiers:
+    repository-code: 'https://github.com/pydantic/pydantic'
+    url: 'https://docs.pydantic.dev/latest/'
+    abstract: >-
+      Pydantic is the most widely used data validation library for Python.
+    keywords:
+      - python
+      - validation
+      - jaon-schema
```
Comment (yezz123) [inline CITATION.cff:39]:
```suggestion
      - json-schema
```

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# YAML 1.2
+---
+authors:
+cff-version: "1.1.0"
```
Comment (Viicos) [inline CITATION.cff:4]: Maybe using the latest spec version (1.2.0 I believe) would be better?

Comment (priya-gitTest) [reply CITATION.cff]: done, please check !!!

File: CITATION.cff [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# YAML 1.2
+---
+abstract: "Pydantic is the most widely used data validation library for Python."
+authors:
+  - family-names: Colvin
+    given-names: Samuel
```
Comment (sydney-runkle) [inline CITATION.cff:6]: Could you add all authors listed here (I think you're missing Alex): https://github.com/pydantic/pydantic/blob/main/pyproject.toml

File: CITATION.cff [hunk 4]
```diff
@@ -0,0 +1,36 @@
+# YAML 1.2
+---
+abstract: "Pydantic is the most widely used data validation library for Python."
+authors:
+  - family-names: Colvin
```
Comment (sydney-runkle) [inline CITATION.cff:5]: I think we can just do a list (don't have to separate family and given name)

File: CITATION.cff (hunk 3 above)
Comment (sydney-runkle) [reply CITATION.cff:6]: There's potentially other info that you could pull from that file as well, looks good overall! Thanks!

**Review by sydney-runkle — CHANGES_REQUESTED** [review]
Looking good overall, requested a few changes, thanks!!

File: CITATION.cff (hunk 4 above)
Comment (priya-gitTest) [reply CITATION.cff]: all authors mentioned here were mentioned. https://docs.pydantic.dev/latest/pydantic_people/

Comment (priya-gitTest) [reply CITATION.cff]: kindly check the latest file .

**Review by priya-gitTest — COMMENTED** [review]
Kindly check. Missing authors also added now.

File: CITATION.cff [hunk 5]
```diff
@@ -0,0 +1,68 @@
+# This CITATION.cff file was generated with cffinit.
+# Visit https://bit.ly/cffinit to generate yours today!
+
+cff-version: 1.2.0
+title: Pydantic
+message: >-
+  If you use this software, please cite it using these
+  metadata.
```
Comment (sydney-runkle) [inline CITATION.cff:8]:
```suggestion
message: 'If you use this software, please cite it as below.'
```

**Review by sydney-runkle — APPROVED** [review]
Looks great, thanks!

**Pull request merged**